src/widgets/label/lv_label.h

```diff
@@ -87,6 +87,15 @@ lv_obj_t * lv_label_create(lv_obj_t * parent);
  */
 void lv_label_set_text(lv_obj_t * obj, const char * text);
```

Comment:
I think this maybe does not need to be its own API. It can be achieved with `lv_label_set_text_fmt`. A comment and a doc update can be all that's needed. What do you think?

```suggestion
 * @note  to set text with a specific length (i.e., not null-terminated), use
          `lv_label_set_text_fmt("%.9s", "only 9 characters of this string");`
          or
          `lv_label_set_text_fmt("%.*s", 9, "only 9 characters of this string");`
 */
void lv_label_set_text(lv_obj_t * obj, const char * text);
```

Reply:
If it's doable with the existing function, we really don't need a new function for that.  @AndyEveritt please confirm if it's working for you too.

Reply:
@liamHowatt I might be wrong but wouldn't that add a reasonable amount of unnecessary overhead to do the string formatting?

Reply:
I have confirmed this works
```cpp
std::string_view sv = /* some non null terminated string */;
lv_label_set_text_fmt(label, "%.*s", (int)text.length(), text.data());
```

My only concern is the potential overhead. It also means you can't do a small optimisation which could be implemented to `lv_label_set_text` where you check if the new text is the same as the current text and don't (re)allocate or free any memory unnecessarily. Currently if you set the text of a label, new heap memory is allocated even if the text hasn't changed

Reply:
With `lv_label_set_text_fmt` it always heap allocates memory to format the string so it would be hard to check if the text is the same as what is already set to prevent an unnecessary allocation

Reply:
I'd accept it as a trade off. If we added `lv_label_set_text_with_length` we should also consider adding the same to span, dropdown, roller, etc causing an API bloat.

Reply:
Ok, so should this PR be closed?

Reply:
We can mention this special formatting  option in the docs. Would you mind updating this PR to have only the docs update in it?

Reply:
I feel like this API should be added. Using `set_text_fmt` feels hacky and `with_length` API for C strings is a very common occurrence in other public C libraries. We can also use only one implementation for both `set_text` and `set_text_with_length` just like this PR proposed

Any chances I might have changed your minds? @liamHowatt @kisvegabor

Reply:
My only concerts are the larger API and code base (for several widget) for a niche issue.

Reply:
I see, in that case I'll close this one for now and maybe get back to it later

Thank you for your contribution @AndyEveritt !